### ISSUE TITLE
[Bug]: `warnAboutOverwriting` does not work on Linux with native windows

### ISSUE THREAD

**Issue by phoracek** [juce-framework/JUCE#1223]
### Detailed steps on how to reproduce the bug

1. Add an extra flag of `FileBrowserComponent::warnAboutOverwriting` to the save dialog example from https://github.com/juce-framework/JUCE/blob/develop/examples/GUI/DialogsDemo.h#L407:
```c++
                fc->launchAsync (FileBrowserComponent::saveMode | FileBrowserComponent::canSelectFiles | FileBrowserComponent::warnAboutOverwriting,
```
2. Compile the example:
```sh
cmake . -B cmake-build -DJUCE_BUILD_EXAMPLES=ON -DJUCE_BUILD_EXTRAS=ON
cmake --build cmake-build --target DialogsDemo
```
3. Run the example:
```sh
cmake-build/examples/GUI/DialogsDemo_artefacts/DialogsDemo
```
4. Check "Use Native Windows"
5. Click "'Save' File Browser"
6. Instead of the dialog popping up, it goes straight to "You picked: <NOTHING>" dialog

### What is the expected behaviour?

When the flag is set, the dialog should pop up (and warn about overwriting if needed).

### Operating systems

Linux

### What versions of the operating systems?

Reported on Fedora 38 with X11 and on openSUSE Tumbleweeed in https://github.com/plugdata-team/plugdata/issues/844.

### Architectures

x86_64

### Plug-in formats (if applicable)

Standalone

### Testing on the `develop` branch

The bug is present on the `develop` branch

### Code of Conduct

- [X] I agree to follow the Code of Conduct

**Comment by KottV** [comment]
I think you're referring to Zenity dialog. Kdialog works fine.
Zenity has dropped `--confirm-overwrite` option with fatal error in 3.91 version,  was fixed in 3.92 https://gitlab.gnome.org/GNOME/zenity/-/issues/55
But since this option is deprecated it seems that it need to be removed from JUCE too.

**Comment by reuk** [comment]
Thanks for reporting, we now check the Zenity version and avoid passing the problematic flag for newer versions:

https://github.com/juce-framework/JUCE/commit/910c60b84a967e8ed2377172a4abe0118b1f782a
